optimum/habana/transformers/trainer.py

```diff
@@ -1763,6 +1763,14 @@ def evaluate(
         num_samples = output.num_samples - self.args.throughput_warmup_steps * total_batch_size
         num_steps = math.ceil(output.num_samples / total_batch_size) - self.args.throughput_warmup_steps
 
+        eval_steps = output.num_samples / total_batch_size
+        if eval_steps <= self.args.throughput_warmup_steps:
+            logger.warning(
+                f" Ignoring warmup steps as number of samples is less to exceed {self.args.throughput_warmup_steps} warmup evaluation steps"
```

Comment:
Not super clear to me. If I understand correctly, what we do here is to actually compute metrics for the warmup steps since there are only warmup steps right?

Reply:
@regisss 
Yes we ignore subtracting the samples processed during warmup stage as this leads to negative number of samples when there are only warmup steps.

Reply:
Yeah, can we put a clearer message?
Something like "Warmup steps are taken into account for the throughput calculation because the number of evaluation steps ({eval_steps}) is smaller than the number of warmup steps({self.args.throughput_warmup_steps})."

Reply:
Thanks, it's very clear! I would just like that you update the message in `logger.warning(...)` because it is not clear there IMO.

Reply:
My Bad, I thought the PR description was unclear. Updated the warning message in trainer and also rounding up the eval_steps to next integer value instead of using a decimal value

Reply:
Thanks!